### PR TITLE
test: added more tests in TokenClaimAirdropTest

### LINKED ISSUE #15070: (CLAIM_34) Given a successful TokenClaimAirdrop transaction having a hollow account with no free maxAutoAssociations slot as receiver should be successful.

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Added: Removed if (nodeStore.keys() != null) check in AddressBookHelper.getNextNodeID and fixed the unit test.

**Related issue(s)**:

Fixes #15070, #15071

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java [hunk 1]
```diff
@@ -1184,6 +1185,46 @@ final Stream<DynamicTest> claimWithSenderAccount() {
...
+        final String ALICE = "ALICE";
+        final String BOB = "BOB";
+        return hapiTest(
+                cryptoCreate(ALICE).balance(ONE_HUNDRED_HBARS),
+                createFT(FUNGIBLE_TOKEN_1, ALICE, 1000L),
+                createHollow(1, i -> BOB),
+                getAliasedAccountInfo(BOB)
+                        .has(accountWith().maxAutoAssociations(-1).hasEmptyKey().noAlias()),
+                cryptoUpdate(BOB).sigMapPrefixes(uniqueWithFullPrefixesFor(BOB)).maxAutomaticAssociations(1),
+                tokenAirdrop(moving(1, FUNGIBLE_TOKEN_1).between(ALICE, BOB)).payingWith(ALICE),
```
Comment (Neeharika-Sompalli) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java:1200]: This record should have an auto association

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java [hunk 2]
```diff
@@ -1184,6 +1185,46 @@ final Stream<DynamicTest> claimWithSenderAccount() {
...
+                // no tokenClaimAirdrop needed here
+                getAccountBalance(BOB).hasTokenBalance(FUNGIBLE_TOKEN_1, 1));
+    }
+
+    @HapiTest
+    @DisplayName("Hollow account should be created before implementation of HIP-904")
```
Comment (Neeharika-Sompalli) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java:1206]: I don't understand this description

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java:1200]: This is CLAIM_34 in this doc, https://limechain.notion.site/Token-Claim-Airdrop-Test-Plan-d1cc52e193b147dabe3232484e7f4d59, set Bob maxAutoAssociations to 1, no no tokenClaimAirdrop needed need after airdrop.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java:1206]: This is CLAIM_35 in above doc. One hollow account without token association and one with it.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java [hunk 3]
```diff
@@ -1184,6 +1185,46 @@ final Stream<DynamicTest> claimWithSenderAccount() {
...
+        final String CAROL = "CAROL";
+        return hapiTest(
+                cryptoCreate(ALICE).balance(ONE_HUNDRED_HBARS),
+                createFT(FUNGIBLE_TOKEN_1, ALICE, 1000L),
+                createHollow(1, i -> BOB),
+                createHollow(1, i -> CAROL),
+                cryptoUpdate(BOB).sigMapPrefixes(uniqueWithFullPrefixesFor(BOB)).maxAutomaticAssociations(0),
+                cryptoUpdate(CAROL)
+                        .sigMapPrefixes(uniqueWithFullPrefixesFor(CAROL))
+                        .maxAutomaticAssociations(0),
```
[the remark this answers is not in the document]
Comment (iwsimon) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java:1219]: Are you saying use this:
`tokenAirdrop(moving(1, FUNGIBLE_TOKEN_1).between(ALICE, BOB)).payingWith(ALICE).sigMapPrefixes(uniqueWithFullPrefixesFor(BOB))`
failed with INVALID_SIGNATURE

**Review by kimbor — APPROVED** [review]

File: gradle/plugins/src/main/kotlin/com.hedera.gradle.application.gradle.kts [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import gradle.kotlin.dsl.accessors._3dd858cc73bb8a89777520d17e15f078.assemble
+import gradle.kotlin.dsl.accessors._3dd858cc73bb8a89777520d17e15f078.jar
+import gradle.kotlin.dsl.accessors._3dd858cc73bb8a89777520d17e15f078.runtimeClasspath
```
Comment (Neeharika-Sompalli) [inline gradle/plugins/src/main/kotlin/com.hedera.gradle.application.gradle.kts:19]: You need to delete these imports. If Intellij doesnt compile you can do `./gradlew --stop` and `/gradlew assemble --rerun-tasks`

Comment (iwsimon) [reply gradle/plugins/src/main/kotlin/com.hedera.gradle.application.gradle.kts]: I removed them. IntelliJ compiles.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @iwsimon

**Pull request merged**